src/data_structures/rmq.rs

```diff
@@ -20,7 +22,7 @@ impl<T: Copy> RMQ<T> {
     /// # Complexity (n = a.len())
     /// - Time: O(n log n)
     /// - Space: O(n log n)
-    pub fn new(a: &[T], op: fn(T, T) -> T) -> Self {
+    pub fn new(a: &[T], op: impl Fn(T, T) -> T + 'static) -> Self {
```

Comment:
I don't love this, but it sounds like we need it?

Introducing `'static` lifetime has caused issues for me in the past, but that's probably cause I never fully understood it, was hoping to avoid haha

Reply:
ya I also don't love it

Reply:
ya it was the first thing that got it compiling

hmm, i'll not merge for now, and keep trying to find smth cleaner